src/models/mod.rs

```diff
@@ -115,3 +117,67 @@ pub(crate) fn create_placeholder(
     item += ")";
     Ok(item)
 }
+
+pub async fn select_retry_or_panic(
+    pool: &sqlx::Pool<sqlx::Postgres>,
+    query: &str,
+    substitution_items: &[String],
+    retry_count: usize,
+) -> anyhow::Result<Vec<sqlx::postgres::PgRow>> {
+    let mut interval = crate::INTERVAL;
+    let mut retry_attempt = 0usize;
+
+    loop {
+        if retry_attempt == retry_count {
+            return Err(anyhow::anyhow!(
+                "Failed to perform query to database after {} attempts. Stop trying.",
+                retry_count
+            ));
+        }
+        retry_attempt += 1;
+
+        let mut args = sqlx::postgres::PgArguments::default();
+        for item in substitution_items {
+            args.add(item);
+        }
+
+        match sqlx::query_with(query, args).fetch_all(pool).await {
+            Ok(res) => return Ok(res),
+            Err(async_error) => {
+                // todo we print here select with non-filled placeholders. It would be better to get the final select statement here
+                tracing::error!(
+                         target: crate::LOGGING_PREFIX,
+                         "Error occurred during {}:\nFailed SELECT:\n{}\n Retrying in {} milliseconds...",
+                         async_error,
+                    query,
+                         interval.as_millis(),
+                     );
+                tokio::time::sleep(interval).await;
+                if interval < crate::MAX_DELAY_TIME {
+                    interval *= 2;
+                }
+            }
+        }
+    }
+}
+
+pub(crate) async fn start_after_interruption(
+    pool: &sqlx::Pool<sqlx::Postgres>,
+) -> anyhow::Result<u64> {
+    // Gets the highest block height from both the nft_events and coin_events table
+    let query = "SELECT MIN(block_height)
+                       FROM (SELECT block_height
+                            FROM   coin_events
+                            UNION ALL
+                            SELECT block_height
+                            FROM   nft_events) AS combined_heights";
```

Comment:
Try to create samples and test it

Reply:
I test this in the prod database  

The query runs on average 134 ms with ~692,980,032 million rows inside of coin_events at the moment. 

https://github.com/near/near-indexer-events/pull/37#discussion_r1048697119

Reply:
I meant functional testing.

I can't ask you to write unit tests here, it's cruel, we don't have any testing environment set up.
Let's imagine we have one.
Which tests do you prefer to cover this function? Could you please draft them?
You don't need to write the code, just create a text draft with the examples, what's the input (from the DB) and what output should we have.